### PR TITLE
Prepare multiplexing support for database plugins

### PR DESCRIPTION
**Pull request by maxcoulombe**
Preparing multiplexing support for all built-in database plugins. Multiplexing is only supported for external plugins so this change is effectively a noop but prepares the field for future projects. As it is a noop for the time being I decided against including a changelog but let me know it you think it'd make sense to include one.

All plugins were tested locally to confirm they still work as expected but share a single process between all plugin instances.

### CODE REVIEW

File: plugins/database/mysql/mysql-database-plugin/main.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"os"
 
 	"github.com/hashicorp/vault/plugins/database/mysql"
-	dbplugin "github.com/hashicorp/vault/sdk/database/dbplugin/v5"
+	"github.com/hashicorp/vault/sdk/database/dbplugin/v5"
```
Comment (vinay-gopalan) [inline plugins/database/mysql/mysql-database-plugin/main.go:8]: Looks like we missed the `ServeMultiplex` operation on the MySQL DB engine

**Review by vinay-gopalan — COMMENTED** [review]
Looking good! Think we might have missed one

**Comment by vinay-gopalan** [comment]
Are we sure that this PR wouldn't warrant a changelog? I think we might have to add one as an `improvement`?

File: plugins/database/mysql/mysql-database-plugin/main.go (hunk 1 above)
Comment (maxcoulombe) [reply plugins/database/mysql/mysql-database-plugin/main.go:8]: Yeah you're right, it's a bit misleading but I did the MySQL project while adding the opt-out flag as this was the system I used to test the feature:
https://github.com/hashicorp/vault/pull/16972/files#diff-aaf53a599c386295cbe7f2ba70be6791e775d5cd51ca8b352e5887f7d8d51b74R24

I can add the same modifications here so the PR is "unified" if someone looks back into the history at some point.

Comment (vinay-gopalan) [reply plugins/database/mysql/mysql-database-plugin/main.go:8]: Ah I see okay I missed that, sorry; I think whatever you prefer is fine. If you'd prefer to keep the MySQL updates in the opt-out PR, I'd suggest adding a link/reference to that code stating that the MySQL updates have been made there. This PR's description might be good spot for that link :)

**Review by vinay-gopalan — APPROVED** [review]
LGTM! Thanks for taking care of these! :)

**Review by austingebauer — APPROVED** [review]

**Pull request merged**